sos/archive.py

```diff
@@ -133,9 +133,10 @@ class FileCacheArchive(Archive):
     _archive_root = ""
     _archive_name = ""
 
-    def __init__(self, name, tmpdir):
+    def __init__(self, name, tmpdir, threads):
```

Comment:
You can't just stuff this into `FileCacheArchive`; it's part of a class hierarchy that defines interfaces: they all need the same initialiser arguments.